dora/core/common/src/main/java/alluxio/underfs/ObjectLowLevelOutputStream.java

```diff
@@ -233,37 +218,44 @@ public void close() throws IOException {
     // Set the closed flag, we never retry close() even if exception occurs
     mClosed = true;
 
-    // Multi-part upload has not been initialized
-    if (!mMultiPartUploadInitialized) {
+//    // Multi-part upload has not been initialized
+//    if (!mMultiPartUploadInitialized) {
+//      if (mFile == null) {
+//        LOG.debug("Streaming upload output stream closed without uploading any data.");
+//        RetryUtils.retry("put empty object for key" + mKey, mMultipartUploader::complete,
+//            mRetryPolicy.get());
+//      } else {
+//        try {
+//          mLocalOutputStream.close();
+//          final String md5 = mHash != null ? Base64.encodeBase64String(mHash.digest()) : null;
+//          RetryUtils.retry("put object for key" + mKey, () -> putObject(mKey, mFile, md5),
+//              mRetryPolicy.get());
+//        } finally {
+//          if (!mFile.delete()) {
+//            LOG.error("Failed to delete temporary file @ {}", mFile.getPath());
+//          }
+//        }
+//      }
+//      return;
+//    }
+
+    try {
+
       if (mFile == null) {
         LOG.debug("Streaming upload output stream closed without uploading any data.");
-        RetryUtils.retry("put empty object for key" + mKey, () -> createEmptyObject(mKey),
+        RetryUtils.retry("put empty object for key" + mKey, mMultipartUploader::complete,
```

Comment:
why do we retry... is the complete operation actually retry-able?

Reply:
yes, we can retry to complete if we don't delete the parts and the last complete op failed.